src/main/java/duke/main/Parser.java

```diff
@@ -0,0 +1,118 @@
+package duke.main;
+
+import duke.exception.EmptyTaskException;
+import duke.exception.InvalidCommandException;
+import duke.exception.InvalidDateException;
+import duke.exception.InvalidIndexException;
+import duke.exception.MissingDateException;
+import duke.task.Task;
+
+public class Parser {
+    private final TaskList taskList;
+    private StringBuilder reply = new StringBuilder();
+
+    Parser(TaskList taskList) {
+        this.taskList = taskList;
+    }
+
+    /**
+     * Interprets user input and executes commands.
+     * The method supports commands for adding, deleting, viewing and completing tasks.
+     * If invalid input is given, users are informed and told to try again.
+     *
+     * @param input User command in console.
+     * @return Output message upon execution of command.
+     */
+    String parse(String input) {
+        try {
+            if (input.equals("list")) {
+                list();
```

Comment:
Would suggest to use this.list(), was abit confused where the method came from when I read it initially